core/vm/logger_trace.go

```diff
@@ -16,11 +14,11 @@ var (
 		CALL:         {traceToAddressCode, traceLastNAddressCode(1), traceCallerProof, traceLastNAddressProof(1)},
 		CALLCODE:     {traceToAddressCode, traceLastNAddressCode(1), traceCallerProof, traceLastNAddressProof(1)},
```

Comment:
what's the different between `traceLastNAddressProof(1)` and doing it in `CaptureEnter`? If they are the same can we remove `traceLastNAddressProof` here?

then change `ExtraData` comments in `core/types/l2trace.go` to 
```
CALL | CALLCODE: [caller contract address’s account state, 
    stack.nth_last(1) (i.e. callee) address’s account state (before called)]
```

Reply:
The data created by traceLastNAddressProof is left for background compatible. I think we should use the account data captured in CaptureEnter, since the balance of called account may be updated and it would only be captured here.